Play services crash caused by NetworkStatsManager.queryDetailsForUidTagState()

```
type: crash
osVersion: google/rango/rango:16/BP4A.260105.004.E1/2026011301:user/release-keys
package: com.google.android.gms:254730035, targetSdk 36
process: com.google.android.gms
processUptime: 74816280 + 640 ms
installer: app.grapheneos.apps
GmsCompatConfig version: 167

gshr: rethrowing exception from lowpool[33]
	at bddw.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):43)
	at android.os.Handler.handleCallback(Handler.java:1070)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at cpny.ms(:com.google.android.gms@254730035@25.47.30 (260400-833691957):1)
	at cpny.dispatchMessage(:com.google.android.gms@254730035@25.47.30 (260400-833691957):140)
	at android.os.Looper.dispatchMessage(Looper.java:333)
	at android.os.Looper.loopOnce(Looper.java:263)
	at android.os.Looper.loop(Looper.java:367)
	at android.app.ActivityThread.main(ActivityThread.java:9331)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:366)
	Suppressed: gjqy: 
		at tk_trace.ScheduledTaskImpl(Unknown Source:0)
		at tk_trace.networktransparency-NetworkDataUpdateTaskBoundService_onRunTaskAsync(Unknown Source:0)
Caused by: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3382)
	at android.os.Parcel.createException(Parcel.java:3366)
	at android.os.Parcel.readException(Parcel.java:3342)
	at android.os.Parcel.readException(Parcel.java:3284)
	at android.net.INetworkStatsService$Stub$Proxy.openSessionForUsageStats(INetworkStatsService.java:481)
	at android.app.usage.NetworkStats.<init>(NetworkStats.java:111)
	at android.app.usage.NetworkStatsManager.queryDetailsForUidTagState(NetworkStatsManager.java:686)
	at android.app.usage.NetworkStatsManager.queryDetailsForUidTagState(NetworkStatsManager.java:648)
	at android.app.usage.NetworkStatsManager.queryDetailsForUidTag(NetworkStatsManager.java:596)
	at dmxb.a(:com.google.android.gms@254730035@25.47.30 (260400-833691957):28)
	at dmwm.gX(:com.google.android.gms@254730035@25.47.30 (260400-833691957):86)
	at iira.w(:com.google.android.gms@254730035@25.47.30 (260400-833691957):11)
	at ijci.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):112)
	at ijqa.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):4)
	at bden.c(:com.google.android.gms@254730035@25.47.30 (260400-833691957):50)
	at bden.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):73)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1154)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:652)
	at bdke.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):8)
	at java.lang.Thread.run(Thread.java:1563)
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10199 not allowed to perform GET_USAGE_STATS
		... 20 more
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.AppOpsManager.noteOp(AppOpsManager.java:9762)
	at android.app.AppOpsManager.noteOp(AppOpsManager.java:9750)
	at android.net.connectivity.android.net.NetworkStatsAccess.hasAppOpsPermission(NetworkStatsAccess.java:207)
	at android.net.connectivity.android.net.NetworkStatsAccess.checkAccessLevel(NetworkStatsAccess.java:146)
	at android.net.connectivity.com.android.server.net.NetworkStatsService.checkAccessLevel(NetworkStatsService.java:1797)
```

The same error happened to me:

````
type: crash
osVersion: google/tokay/tokay:16/BP4A.260105.004.E1/2026011301:user/release-keys
flags: dev options enabled
package: com.google.android.gms:254730035, targetSdk 36
process: com.google.android.gms
processUptime: 2140 + 539 ms
installer: app.grapheneos.apps
GmsCompatConfig version: 167

gshr: rethrowing exception from lowpool[2]
	at bddw.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):43)
	at android.os.Handler.handleCallback(Handler.java:1070)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at cpny.ms(:com.google.android.gms@254730035@25.47.30 (260400-833691957):1)
	at cpny.dispatchMessage(:com.google.android.gms@254730035@25.47.30 (260400-833691957):140)
	at android.os.Looper.dispatchMessage(Looper.java:333)
	at android.os.Looper.loopOnce(Looper.java:263)
	at android.os.Looper.loop(Looper.java:367)
	at android.app.ActivityThread.main(ActivityThread.java:9331)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:366)
	Suppressed: gjqy: 
		at tk_trace.ScheduledTaskImpl(Unknown Source:0)
		at tk_trace.networktransparency-NetworkDataUpdateTaskBoundService_onRunTaskAsync(Unknown Source:0)
Caused by: java.lang.SecurityException: com.google.android.gms from uid 10197 not allowed to perform GET_USAGE_STATS
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3382)
	at android.os.Parcel.createException(Parcel.java:3366)
	at android.os.Parcel.readException(Parcel.java:3342)
	at android.os.Parcel.readException(Parcel.java:3284)
	at android.net.INetworkStatsService$Stub$Proxy.openSessionForUsageStats(INetworkStatsService.java:481)
	at android.app.usage.NetworkStats.<init>(NetworkStats.java:111)
	at android.app.usage.NetworkStatsManager.queryDetailsForUidTagState(NetworkStatsManager.java:686)
	at android.app.usage.NetworkStatsManager.queryDetailsForUidTagState(NetworkStatsManager.java:648)
	at android.app.usage.NetworkStatsManager.queryDetailsForUidTag(NetworkStatsManager.java:596)
	at dmxb.a(:com.google.android.gms@254730035@25.47.30 (260400-833691957):28)
	at dmwm.gX(:com.google.android.gms@254730035@25.47.30 (260400-833691957):86)
	at iira.w(:com.google.android.gms@254730035@25.47.30 (260400-833691957):11)
	at ijci.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):112)
	at ijqa.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):4)
	at bden.c(:com.google.android.gms@254730035@25.47.30 (260400-833691957):50)
	at bden.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):73)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1154)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:652)
	at bdke.run(:com.google.android.gms@254730035@25.47.30 (260400-833691957):8)
	at java.lang.Thread.run(Thread.java:1563)
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10197 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10197 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10197 not allowed to perform GET_USAGE_STATS
		... 20 more
	Suppressed: java.lang.SecurityException: com.google.android.gms from uid 10197 not allowed to perform GET_USAGE_STATS
		... 20 more
```
`

--- a/framework-t/src/android/app/usage/NetworkStatsManager.java
+++ b/framework-t/src/android/app/usage/NetworkStatsManager.java
@@ -31,6 +31,7 @@ import android.annotation.RequiresPermission;
 import android.annotation.SystemApi;
 import android.annotation.SystemService;
 import android.annotation.WorkerThread;
+import android.app.compat.gms.GmsCompat;
 import android.app.usage.NetworkStats.Bucket;
 import android.compat.annotation.UnsupportedAppUsage;
 import android.content.Context;
@@ -593,8 +594,19 @@ public class NetworkStatsManager {
     @WorkerThread
     public NetworkStats queryDetailsForUidTag(int networkType, @Nullable String subscriberId,
             long startTime, long endTime, int uid, int tag) throws SecurityException {
-        return queryDetailsForUidTagState(networkType, subscriberId, startTime, endTime, uid,
-            tag, NetworkStats.Bucket.STATE_ALL);
+        try {
+            return queryDetailsForUidTagState(networkType, subscriberId, startTime, endTime, uid,
+                tag, NetworkStats.Bucket.STATE_ALL);
+        } catch (SecurityException e) {
+            if (GmsCompat.isEnabled()) {
+                // GmsCore null-checks the return value despite the method being marked as non-null.
+                // An alternative to returning null is to make a stub instance of NetworkStats, but
+                // it's complex because of the way NetworkStats is implemented.
+                //noinspection DataFlowIssue
+                return null;
+            }
+            throw e;
+        }
     }
 
     /**

